doctest/parts/doctest_fwd.h

```diff
@@ -576,12 +578,21 @@ class DOCTEST_INTERFACE String
 
     int compare(const char* other, bool no_case = false) const;
     int compare(const String& other, bool no_case = false) const;
+    const char *compare(const Contains& other) const;
```

Comment:
Why does this return a `const char*`? Imo this should be a method on `Contains` (e.g.: `Contains::checkWith(String)`) that simply returns a bool. This will also make the `Contains` forward declaration superfluous and make the dependency between `String` and `Contains` unidirectional.